modules/gateway/gateway.go

```diff
@@ -56,62 +56,33 @@ func (g *Gateway) Bootstrap(bootstrapPeer network.Address) (err error) {
 }
 
 // RelayBlock relays a block, both locally and to the network.
-// RelayBlock is called by Miners.
 func (g *Gateway) RelayBlock(b consensus.Block) (err error) {
 	err = g.state.AcceptBlock(b)
 	if err != nil {
 		return
 	}
-	g.broadcast("RelayBlock", b, nil)
-	return
-}
-
-// RelayTransaction relays a transaction, both locally and to the network.
-// RelayTransaction is called by Wallets.
-func (g *Gateway) RelayTransaction(t consensus.Transaction) (err error) {
-	err = g.tpool.AcceptTransaction(t)
-	if err != nil {
-		return
-	}
-	g.broadcast("RelayTransaction", t, nil)
-	return
-}
 
-// SharePeers returns up to 10 randomly selected peers.
-func (g *Gateway) SharePeers() (peers []network.Address, err error) {
-	g.mu.RLock()
-	defer g.mu.RUnlock()
-	for peer := range g.peers {
-		if len(peers) > 10 {
-			return
+	// Check if b is in the current path.
+	height, exists := g.state.HeightOfBlock(b.ID())
```

Comment:
maybe a `IsInCurrentPath` helper is warranted...would that be useful anywhere else?